Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -253,29 +253,81 @@ EnableLegacyRegions (
 
 
 /**
-  Enable SMI on S3 resume path
+  Restore registers saved during normal boot
 
 **/
 VOID
-EnableSmi (
+RestoreS3Regs (
+  VOID
+)
+{
+  S3_SAVE_REG           *S3SaveReg;
+  UINT8                 Index;
+  UINT32                Data32;
+  UINT16                Data16;
+
+  S3SaveReg = (S3_SAVE_REG *) (MmioRead32 (TO_MM_PCI_ADDRESS (0x00000000) + TSEG) & ~0xF);
```

Comment:
you need to verify the SaSaveReg pointer is valid.   Check for NULL, and also verify the signature S3_SAVE_REG_SIG.

Reply:
Added the checks.